### PR TITLE
Add notes to CSV Export. Move save folder to Downloads. Perform additional checks before saving. Fix bug with mmol. Closes: #290.

### PR DESCRIPTION
**Pull request by PaoloRotolo**
Please review

### CODE REVIEW

File: app/src/main/java/org/glucosio/android/presenter/ExportPresenter.java [hunk 1]
```diff
@@ -130,6 +138,21 @@ protected void onPostExecute(String filename) {
         }
     }
 
+    private boolean dirExists() {
+        final File file = new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS) + "/glucosio");
+        return file.exists() || file.mkdirs();
+    }
+
+    private boolean arePermissionsGranted() {
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/presenter/ExportPresenter.java]: Not used method?

File: app/src/main/java/org/glucosio/android/tools/ReadingToCSV.java [hunk 2]
```diff
@@ -42,72 +42,84 @@ public ReadingToCSV(Context mContext) {
...
 
+                // Concentration | Measured | Date | Time | Notes | Unit of Measurement
                 if ("mg/dL".equals(um)) {
                     for (int i = 0; i < readings.size(); i++) {
+                        GlucoseReading reading = readings.get(i);
+                        osw.append(dateTool.convertRawDate(reading.getCreated() + ""));
+                        osw.append(',');
 
-                        osw.append(readings.get(i).getReading() + "mg/dL");
+                        osw.append(dateTool.convertRawTime(reading.getCreated() + ""));
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/tools/ReadingToCSV.java]: Printing date time two times?

File: app/src/main/java/org/glucosio/android/tools/ReadingToCSV.java [hunk 3]
```diff
@@ -42,72 +42,84 @@ public ReadingToCSV(Context mContext) {
...
                         osw.append(',');
 
-                        osw.append(readings.get(i).getReading_type() + "");
+                        osw.append(reading.getReading() + "");
                         osw.append(',');
 
-                        osw.append(dateTool.convertRawDate(readings.get(i).getCreated() + ""));
+                        osw.append("mg/dL");
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/tools/ReadingToCSV.java]: Are we really taking it in mg/dL? Or we actually store it in different formats?

**Comment by emartynov** [comment]
Just minor comments. And I don't understand the fix again.

File: app/src/main/java/org/glucosio/android/tools/ReadingToCSV.java (hunk 2 above)
Comment (PaoloRotolo) [inline app/src/main/java/org/glucosio/android/tools/ReadingToCSV.java]: The first one converts the date. The second one the time:
```
osw.append(dateTool.convertRawDate(reading.getCreated() + ""));
osw.append(dateTool.convertRawTime(reading.getCreated() + ""));
```

**Comment by PaoloRotolo** [comment]
So there were a typo in the CSV class when exporting with mmol as unit of measurement. The result was a broken CSV table. I fixed the typo and added new fields for Notes and other info. Also now the exported file goes in the Download folder (since on some devices we haven't permission to write outside).

File: app/src/main/java/org/glucosio/android/presenter/ExportPresenter.java (hunk 1 above)
Comment (PaoloRotolo) [inline app/src/main/java/org/glucosio/android/presenter/ExportPresenter.java]: Remove, thanks.

**Pull request merged**